cscore/src/test/java/edu/wpi/first/cscore/FrameTimeTest.java

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package edu.wpi.first.cscore;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import edu.wpi.first.cscore.raw.RawSink;
+import edu.wpi.first.cscore.raw.RawSource;
+import edu.wpi.first.util.PixelFormat;
+import edu.wpi.first.util.RawFrame;
+import edu.wpi.first.util.TimestampSource;
+import org.junit.jupiter.api.Test;
+
+public class FrameTimeTest {
+  @Test
+  public void testFrameTimeRoundTrip() {
+    // Given a Source
+    RawSource source = new RawSource("foobar", PixelFormat.kBGR, 320, 240, 30);
+
+    // And a sink connected to it
+    RawSink sink = new RawSink("foobar2");
+    sink.setSource(source);
+
+    // And as new frame with a non-zero time
+    RawFrame frame = new RawFrame();
+    frame.setTimeInfo(12, TimestampSource.kV4lEoF);
+
+    // And a sink waiting for a new frame to arrive
+    RawFrame sinkFrame = new RawFrame();
+    Thread sinkGrabber =
+        new Thread(
+            () -> {
+              long ret = sink.grabFrame(sinkFrame);
+              System.err.println(ret);
+              assertTrue(ret > 0);
+            });
+    sinkGrabber.start();
+
+    // When a new frame is added to the source
+    // (Note: the frame time and time source are overridden by RawSourceImpl, and changing this is
+    // spooky)
```

Comment:
Could consider updating cscore to make this more testable, but i'm happy with this for now.